### PR TITLE
golangci-lint improvements

### PR DESCRIPTION
**Pull request by imiric**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
           name: Run golangci-lint
           command: |
             export PATH="$GOPATH/bin:$PATH"
-            basecommit=$(git log --first-parent --pretty=format:'%H %D' | egrep '/(master|new-schedulers)' | head -1 | cut -d' ' -f1)
+            basecommit=$(git log --decorate | grep '^commit.*origin/' | sed -n 2p | cut -d' ' -f2)
```
Comment (imiric) [inline .circleci/config.yml:56]: I think this will fail if a specific commit is checked out and not a branch... Let me test force pushing.

Comment (imiric) [reply .circleci/config.yml:56]: No, works as expected. Still, this feels super hacky, so I wouldn't be surprised if it fails at some point. There doesn't seem to be a 100% reliable solution. :-/

Comment (na--) [reply .circleci/config.yml:56]: If needed, we just use `master`. After all, the goal is for a continuous improvement of the linter state of `master`.

Comment (na--) [reply .circleci/config.yml:56]: In any case, this is good enough for now, we can always change it if we see problems

**Pull request merged**